samples/bluetooth/handsfree_ag/src/main.c

```diff
@@ -0,0 +1,386 @@
+			err = bt_hfp_ag_connect(conn, &hfp_ag, value);
+			if (err != 0) {
+				printk("Fail to create hfp AG connection (err %d)\n", err);
+			}
+		}
+	}
+
+	return BT_SDP_DISCOVER_UUID_STOP;
+}
+
+static struct bt_sdp_discover_params sdp_discover = {
+	.func = sdp_discover_cb,
+	.pool = &sdp_discover_pool,
+	.uuid = BT_UUID_DECLARE_16(BT_SDP_HANDSFREE_SVCLASS),
+};
+
+static void connected(struct bt_conn *conn, uint8_t err)
+{
+	int res;
+
+	if (err) {
+		if (default_conn != NULL) {
+			default_conn = NULL;
+		}
+		printk("Connection failed (err 0x%02x)\n", err);
+	} else {
+		if (default_conn == conn) {
+			struct bt_conn_info info;
+
+			bt_conn_get_info(conn, &info);
+			if (info.type != BT_CONN_TYPE_BR) {
+				return;
+			}
+
+			/*
+			 * Do an SDP Query on Successful ACL connection complete with the
+			 * required device
+			 */
+			res = bt_sdp_discover(default_conn, &sdp_discover);
+			if (res) {
+				printk("SDP discovery failed (err %d)\r\n", res);
+			} else {
+				printk("SDP discovery started\r\n");
+			}
+			printk("Connected\n");
+		}
+	}
+}
+
+static void disconnected(struct bt_conn *conn, uint8_t reason)
+{
+	printk("Disconnected (reason 0x%02x)\n", reason);
+
+	if (default_conn != conn) {
+		return;
+	}
+
+	if (default_conn) {
+		default_conn = NULL;
+	} else {
+		return;
+	}
+}
+
+static void security_changed(struct bt_conn *conn, bt_security_t level, enum bt_security_err err)
+{
+	char addr[BT_ADDR_LE_STR_LEN];
+	struct bt_conn_info info;
+
+	bt_conn_get_info(conn, &info);
+
+	bt_addr_to_str(info.br.dst, addr, sizeof(addr));
+
+	printk("Security changed: %s level %u (err %d)\n", addr, level, err);
+}
+
+static struct bt_conn_cb conn_callbacks = {
+	.connected = connected,
+	.disconnected = disconnected,
+	.security_changed = security_changed,
+};
+
+static void scan_discovery_cb(struct bt_br_discovery_result *results, size_t count)
+{
+	char addr[BT_ADDR_LE_STR_LEN];
+	uint8_t *eir;
+	bool flag = true;
+	static uint8_t temp[240];
+	size_t len = sizeof(results->eir);
+	size_t major_device;
+	size_t minor_device;
+	size_t i;
+
+	for (i = 0; i < count; i++) {
+		bt_addr_to_str(&results[i].addr, addr, sizeof(addr));
+		printk("Device[%d]: %s, rssi %d, cod 0x%X%X%X", i, addr, results[i].rssi,
+		       results[i].cod[0], results[i].cod[1], results[i].cod[2]);
+
+		major_device = BT_COD_MAJOR_DEVICE_CLASS(results[i].cod);
+		minor_device = BT_COD_MINOR_DEVICE_CLASS(results[i].cod);
+
+		if ((major_device & BT_COD_MAJOR_AUDIO_VIDEO) &&
+		    (minor_device & BT_COD_MAJOR_AUDIO_VIDEO_MINOR_HANDS_FREE)) {
+			flag = false;
+		}
+
+		eir = results[i].eir;
+
+		while ((eir[0] > 2) && (len > eir[0])) {
+			switch (eir[1]) {
+			case BT_DATA_NAME_SHORTENED:
+			case BT_DATA_NAME_COMPLETE:
+				memcpy(temp, &eir[2], eir[0] - 1);
+				temp[eir[0] - 1] = '\0'; /* Set end flag */
+				printk(", name %s", temp);
+				break;
+			}
+			len = len - eir[0] - 1;
+			eir = eir + eir[0] + 1;
+		}
+		printk("\n");
+
+		if (flag == false) {
```

Comment:
`if (!flag)`, however `flag` is not a good name (too generic). Maybe call it something like `cod_hf`.

Reply:
updated